### PR TITLE
DO-251: Fixed issue with Date dropdown.

### PR DESCRIPTION
**Pull request by dhruvlahoti**

### CODE REVIEW

File: src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx [hunk 1]
```diff
@@ -14,45 +16,37 @@ import Input from '../../../Input';
...
-    let day = `${d.getDate()}`;
-    const year = d.getFullYear();
+    const dateObj = new Date(date);
+    let m = `${dateObj.getMonth() + 1}`;
+    let d = `${dateObj.getDate()}`;
+    const y = dateObj.getFullYear();
 
-    if (month.length < 2) month = `0${month}`;
-    if (day.length < 2) day = `0${day}`;
+    if (m.length < 2) m = `0${m}`;
```
Comment (mohitsaladiya) [inline src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx:39]: Please mention in the comments what is the significance of comparing it with length 2

File: src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx [hunk 2]
```diff
@@ -14,45 +16,37 @@ import Input from '../../../Input';
...
-
-  const placeholder = get(uiSchema, 'ui:placeholder', '');
-
   const formatDate = (date) => {
-    const d = new Date(date);
-    let month = `${d.getMonth() + 1}`;
-    let day = `${d.getDate()}`;
-    const year = d.getFullYear();
+    const dateObj = new Date(date);
+    let m = `${dateObj.getMonth() + 1}`;
```
Comment (mohitsaladiya) [inline src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx:35]: Please mention in the comments why are we adding 1 in the month here in the comments.

File: src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx [hunk 3]
```diff
@@ -14,45 +16,37 @@ import Input from '../../../Input';
...
+    let d = `${dateObj.getDate()}`;
+    const y = dateObj.getFullYear();
```
Comment (mohitsaladiya) [inline src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx:37]: Revert the variables with their names as per earlier implementation.

File: src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx (hunk 1 above)
Comment (dhruvlahoti) [reply src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx]: Done

File: src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx (hunk 2 above)
Comment (dhruvlahoti) [reply src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx]: Done

File: src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx (hunk 3 above)
Comment (dhruvlahoti) [reply src/components/common/Typography/Form/FormFields/DatePickerInput/DatePickerInput.jsx]: Done

**Pull request merged**